keps/sig-storage/2485-read-write-once-pod-pv-access-mode/README.md

```diff
@@ -0,0 +1,1024 @@
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+In order to upgrade a cluster to use this feature, the user will need to restart
+the kube-apiserver, kube-controller-manager, kube-scheduler, and kubelet with
+the ReadWriteOncePod feature gate enabled.  Additionally they will need to
+update their CSI drivers and sidecars to versions that depend on the new
+Kubernetes API and CSI spec.
+
+When downgrading a cluster to disable this feature, the user will need to
+restart the kube-apiserver with the ReadWriteOncePod feature gate disabled. When
+disabling this feature gate, any existing volumes with the ReadWriteOncePod
+access mode will continue to exist, but can only be deleted. An alternative is
+to allow these volumes to be treated as ReadWriteOnce, however that would
+violate the intent of the user and so it is not recommended.
+
+If a user downgrades their CSI drivers or sidecars, any existing volumes using
+ReadWriteOnce should continue working (switching from `SINGLE_NODE_MULTI_WRITER`
+to `SINGLE_NODE_WRITER`). This behavior is ultimately up to each CSI driver, but
+they should be designed with this backwards compatibility in mind.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+
+#### API Server Version N / Scheduler Version N / Kubelet Version N-1 or N-2
+
+When starting two pods with both using the same PVC with ReadWriteOncePod, one pod
+will successfully start, but the other will not be scheduled due to the
+ReadWriteOncePod access mode conflict.
+
+When starting the same two pods but also setting `pod.spec.nodeName` to the same
+node, kubelet will not enforce the access mode and will proceed with starting
+both pods.
+
+For older kubelets, [ReadWriteOncePod will map to access mode `UNKNOWN`]. How this
+access mode is used will vary across CSI drivers.
+
+[ReadWriteOncePod will map to access mode `UNKNOWN`]: https://github.com/kubernetes/kubernetes/blob/v1.21.0/pkg/volume/csi/csi_client.go#L512
+
+#### API Server Version N / Scheduler Version N-1 / Kubelet Version N
```

Comment:
I believe the "kubelet N" is not valid. The entire control plane needs to be upgraded before nodes can.

Reply:
Updated to use kubelet N-1 and N-2